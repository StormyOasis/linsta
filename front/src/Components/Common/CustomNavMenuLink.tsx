import React, { ReactNode } from "react";
import { Link } from "react-router-dom";
import { styled } from "styled-components";
import * as styles from '../Layout/Main/Main.module.css';
import { Div, Flex, Span } from "./CombinedStyling";

const InnerNavLinkWrapper = styled(Div)`
    padding: 12px;
    margin: 2px 0;
    display: inline-flex;
    cursor: pointer;
    border-radius: 8px;

    &:hover {
        background-color: ${(props) => props.theme['colors'].navLinkHoverColor};
    }        


    @media (min-width: ${props => props.theme["breakpoints"].md}px) and 
        (max-width: ${props => props.theme["breakpoints"].lg - 1}px) {
        padding-top: 9px;
    }

    @media (min-width: ${props => props.theme["breakpoints"].lg - 1}px) {
        width: 100%;
        align-items: center;
        flex-direction: row;
    }
`;

const NavLink = styled(Link)`
    text-decoration: none;
    color: ${props => props.theme["colors"].navLinkTextColor};

    display: flex;
    flex-basis: 100%;
    justify-content: center;
`;

const DivLink = styled(Flex)`
    text-decoration: none;
    color: ${props => props.theme["colors"].navLinkTextColor};
    flex-basis: 100%;
    justify-content: center;
`;

interface CustomNavMenuLinkProps {
    to: string | null;
    text: string;
    iconElement: ReactNode;
    paddingLeft?: number;
    onClick?: () => void;
    matchesLargestBP: boolean;
};

const CustomNavMenuLink: React.FC<CustomNavMenuLinkProps> = React.memo((props: CustomNavMenuLinkProps) => {
    const LinkContents: ReactNode =
        <InnerNavLinkWrapper>
            <Div className={styles.iconWrapper}>
                {props.iconElement}
            </Div>
            {props.matchesLargestBP &&
                <Div $paddingLeft={`${props.paddingLeft}px`}>
                    <Span className={styles.text}>{props.text}</Span>
                </Div>
            }
        </InnerNavLinkWrapper>;

    if (props.to != null) {
        return (
            <NavLink to={props.to} onClick={props.onClick} aria-label={props.text}>
                {LinkContents}
            </NavLink>
        );
    }

    return (
        <DivLink onClick={props.onClick} aria-description={props.text}>
            {LinkContents}
        </DivLink>
    );
});

export default CustomNavMenuLink;
