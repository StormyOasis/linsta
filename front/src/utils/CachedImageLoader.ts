import { clear, del, get, set } from 'idb-keyval';
import { blobToBase64 } from "./utils";

export const loadImageCached = async (id: string, url: string): Promise<string> => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // Image not in the cache so have to fetch it
        try {
            const result = await fetch(url);
            if(result.status === 200) {
                const blob = await result.blob();
                const b64:string = await blobToBase64(blob) as string;

                // add the base64 data into the cache and return it
                await set(id, b64);

                return b64;
            }
        } catch(err) {
            console.log(err);
            return url; //default to what was passed in
        }

    } else {
        // Image is in the cache already as a base64 string, use that as is
        return value;
    }

    return url; //default to what was passed in
}

export const setImage = async (id: string, data: string) => {     
    await set(id, data);
}

export const deleteFromCache = async (id: string) => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // key not present in cache so do nothing
        return;
    }

    // delete from the cache
    await del(id);
}

export const clearCache = async () => {
    await clear();
}