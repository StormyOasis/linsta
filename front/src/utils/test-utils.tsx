import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };