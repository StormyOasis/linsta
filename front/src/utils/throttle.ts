import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;