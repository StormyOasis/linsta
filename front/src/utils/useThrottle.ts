import { useRef, useCallback } from 'react';

/**
 * React hook for throttling a callback function.
 * The callback will only be invoked at most once every `delay` milliseconds.
 *
 * @param callback - The function to throttle.
 * @param delay - The minimum delay (ms) between calls.
 * @returns A throttled version of the callback.
 */
const useThrottle = <T extends (...args: any[]) => void>(callback: T, delay: number) => {
    const lastCall = useRef<number>(0);

    return useCallback(
        (...args: Parameters<T>) => {
            const now = performance.now();
            if (now - lastCall.current >= delay) {
                lastCall.current = now;
                callback(...args);
            }
        },
        [callback, delay]
    );
};

export default useThrottle;