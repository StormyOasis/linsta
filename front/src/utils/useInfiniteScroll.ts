import React, { useEffect, useRef } from 'react';
import useThrottle from './useThrottle';

/**
 * Type for the async load more function.
 */
type AsyncLoadMoreFunction = () => Promise<void>;

/**
 * Custom hook for infinite scrolling.
 * Attaches a throttled scroll event listener to a container and triggers `loadMore`
 * when the user scrolls near the bottom.
 *
 * @param loadMore - Async function to load more items.
 * @param isLoading - Whether a load is currently in progress.
 * @param containerRef - Ref to the scrollable container element.
 */
const useInfiniteScroll = (loadMore: AsyncLoadMoreFunction, 
    isLoading: boolean, 
    childRef: React.MutableRefObject<HTMLDivElement | null>) => {
        
    const hasScrolled = useRef<boolean>(false);  // Track if the user has scrolled

    // Throttled scroll handler
    const throttledHandleScroll = useThrottle(async () => {
        if (typeof window !== 'undefined' && childRef.current) {
            const element = childRef.current as HTMLElement;
            const currentScroll = window.innerHeight + element.scrollTop;

            if (currentScroll + 256 >= element.scrollHeight && !isLoading && !hasScrolled.current) {
                hasScrolled.current = true;  // Mark as scrolled at least once
                await loadMore();
            }
        }
    }, 200);

    useEffect(() => {
        const childElement = childRef.current;
        if (childElement) {
            childElement.addEventListener('scroll', throttledHandleScroll);
        }

        // Clean up event listener on unmount
        return () => {
            if (childElement) {
                childElement.removeEventListener('scroll', throttledHandleScroll);
            }
        };
    }, [throttledHandleScroll, childRef]);

    // Reset trigger flag when loading finishes
    useEffect(() => {
        if (!isLoading) {
            hasScrolled.current = false;
        }
    }, [isLoading]);
}

export default useInfiniteScroll;