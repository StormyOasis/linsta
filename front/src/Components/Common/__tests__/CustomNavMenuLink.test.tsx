import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import CustomNavMenuLink from "../CustomNavMenuLink";
import { renderWithStore } from "../../../utils/test-utils";
import '@testing-library/jest-dom';

describe("CustomNavMenuLink", () => {
    const icon = <svg data-testid="icon" />;
    const text = "Menu Link";
    const paddingLeft = 24;

    it("renders NavLink when 'to' prop is provided", () => {
        const r = renderWithStore(
            <CustomNavMenuLink
                to="/test"
                text={text}
                iconElement={icon}
                paddingLeft={paddingLeft}
                matchesLargestBP={true}
            />
        );
        // Should render a link with correct aria-label and text
        const link = screen.getByRole("link", { name: text });
        expect(link).toBeInTheDocument();
        expect(screen.getByTestId("icon")).toBeInTheDocument();
        expect(screen.getByText(text)).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("renders DivLink when 'to' prop is null", () => {
        const r = renderWithStore(
            <CustomNavMenuLink
                to={null}
                text={text}
                iconElement={icon}
                paddingLeft={paddingLeft}
                matchesLargestBP={true}
            />
        );
        // Should render a div with aria-description and text
        const div = screen.getByText(text).closest("div");
        expect(div).toBeInTheDocument();
        expect(screen.getByTestId("icon")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("calls onClick when clicked (NavLink)", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(
                <CustomNavMenuLink
                    to="/test"
                    text={text}
                    iconElement={icon}
                    paddingLeft={paddingLeft}
                    matchesLargestBP={true}
                    onClick={handleClick}
                />
        );
        const link = screen.getByRole("link", { name: text });
        fireEvent.click(link);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot()
    });

    it("calls onClick when clicked (DivLink)", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(
            <CustomNavMenuLink
                to={null}
                text={text}
                iconElement={icon}
                paddingLeft={paddingLeft}
                matchesLargestBP={true}
                onClick={handleClick}
            />
        );
        const div = screen.getByText(text).closest("div");
        fireEvent.click(div!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("does not render text when matchesLargestBP is false", () => {
        const r = renderWithStore(
            <CustomNavMenuLink
                to={null}
                text={text}
                iconElement={icon}
                paddingLeft={paddingLeft}
                matchesLargestBP={false}
            />
        );
        expect(screen.queryByText(text)).not.toBeInTheDocument();
        expect(screen.getByTestId("icon")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });
});