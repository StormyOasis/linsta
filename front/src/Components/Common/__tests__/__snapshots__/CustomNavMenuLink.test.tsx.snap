// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNavMenuLink calls onClick when clicked (DivLink) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-description="Menu Link"
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <svg
              data-testid="icon"
            />
          </div>
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
            >
              Menu Link
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-description="Menu Link"
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <svg
            data-testid="icon"
          />
        </div>
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
        >
          <span
            class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
          >
            Menu Link
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`CustomNavMenuLink calls onClick when clicked (NavLink) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Menu Link"
        class="CustomNavMenuLink__NavLink-sc-1cwq39-1 icZHz"
        href="/test"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <svg
              data-testid="icon"
            />
          </div>
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
            >
              Menu Link
            </span>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Menu Link"
      class="CustomNavMenuLink__NavLink-sc-1cwq39-1 icZHz"
      href="/test"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <svg
            data-testid="icon"
          />
        </div>
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
        >
          <span
            class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
          >
            Menu Link
          </span>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`CustomNavMenuLink does not render text when matchesLargestBP is false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-description="Menu Link"
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <svg
              data-testid="icon"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-description="Menu Link"
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <svg
            data-testid="icon"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`CustomNavMenuLink renders DivLink when 'to' prop is null 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-description="Menu Link"
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <svg
              data-testid="icon"
            />
          </div>
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
            >
              Menu Link
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-description="Menu Link"
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CustomNavMenuLink__DivLink-sc-1cwq39-2 escqfF ckFvxY eiXdbS"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <svg
            data-testid="icon"
          />
        </div>
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
        >
          <span
            class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
          >
            Menu Link
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`CustomNavMenuLink renders NavLink when 'to' prop is provided 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Menu Link"
        class="CustomNavMenuLink__NavLink-sc-1cwq39-1 icZHz"
        href="/test"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
          >
            <svg
              data-testid="icon"
            />
          </div>
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
          >
            <span
              class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
            >
              Menu Link
            </span>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Menu Link"
      class="CustomNavMenuLink__NavLink-sc-1cwq39-1 icZHz"
      href="/test"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CustomNavMenuLink__InnerNavLinkWrapper-sc-1cwq39-0 escqfF kIiyXc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
        >
          <svg
            data-testid="icon"
          />
        </div>
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 htdXAs"
        >
          <span
            class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
          >
            Menu Link
          </span>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
